rebasehelper/logger.py

```diff
@@ -77,11 +77,39 @@ def add_file_handler(logger_object, path, formatter_object=None, level=None):
             file_handler.setFormatter(formatter_object)
         logger_object.addHandler(file_handler)
 
+    @staticmethod
+    def add_logging_level(level_name, level_number, method_name=None):
+        if method_name is None:
+            method_name = level_name.lower()
+
+        if hasattr(logging, level_name):
+            raise AttributeError('{} already defined in logging module'.format(level_name))
+        if hasattr(logging, method_name):
+            raise AttributeError('{} already defined in logging module'.format(method_name))
+        if hasattr(logging.getLoggerClass(), method_name):
+            raise AttributeError('{} already defined in logger class'.format(method_name))
+
+        def log_level(self, message, *args, **kwargs):
+            if self.isEnabledFor(level_number):
+                self._log(level_number, message, args, **kwargs)  # pylint: disable=protected-access
+
+        logging.addLevelName(level_number, level_name)
+        setattr(logging, level_name, level_number)
+        setattr(logging.getLoggerClass(), method_name, log_level)
```

Comment:
I don't like this at all. Why won't you define a custom class based on `Logger` and register it using `setLoggerClass()`?